woocommerce/woocommerce
# Add cherry pick tool workflow

### All Submissions:






### Changes proposed in this Pull Request:



Closes #34135 

This PR automates the process of creating two PRs that otherwise the release lead would have to manually do. When a PR from product teams are created and merged that targets an outgoing release that needs to be cherry picked, it will automatically create two PRs for this. The triggering flag is having the milestone tagged on the PR and that the PR is merged.

Alternatively, we can also manually run this process in the workflow manual trigger.

### How to test the changes in this Pull Request:

1. Fork this repo. All the steps here on out will be done in your fork ONLY!
2. Enable Actions ( by default it is disabled ). `settings/actions`.
3. Enable Issues `settings`.
4. Ensure the repo is setup as squash and merge only.
5. Create a milestone named `6.8`.
6. In trunk, merge this branch (add/cherry-pick-workflow-con)
7. Create a branch named `release/6.8` on your Forked GitHub repo.
8. Update `plugins/woocommerce/readme.txt` so that it contains changelog entries. You can copy from https://github.com/woocommerce/woocommerce/blob/release/6.8/plugins/woocommerce/readme.txt
9. Create a private Slack channel. Right click on the channel and click on "Channel Details". Note the channel ID at the bottom. Save that information.
10. Create an action secrets token named `CODE_FREEZE_BOT_TOKEN`. You can get the bot token from the Slack APP you were invited to from your email.
11. Create an action secrets token named `WOO_RELEASE_SLACK_CHANNEL`. Add the channel ID you saved from step 4.
12.  Create a branch from trunk name `testing-cherry`.
13. Make some changes to `plugins/woocommerce/woocommerce.php` and add a changelog file in `plugins/woocommerce/changelog`.
14. Commit the changes and push this branch up to your repo and create PRs for it. BECAREFUL, make sure you select your own fork when creating the PR on GitHub because sometimes it defaults to the upstream origin.
15. Merge the PR without doing anything else. Witness that the cherry pick automation fires but it stops as far as the verification step. This is expected because no milestone was added.
16. Repeat steps 12 - 14. This time tag it with 6.8 milestone. Squash and merge it. Note down the PR number for later test.
17. Witness the cherry pick tool automation runs and this time it should have created two PRs. One that targets the release branch (`release/6.8`). Another PR targets the trunk branch with the deleted changelog file. Also check your Slack channel that you created from step 8. You should see a message from the code freeze notification bot with the listed PRs for the release lead to review.
18. Closed both of these auto generated PRs and delete the branches it created.
19. Go to your actions workflow -> `Cherry Pick Tool`.
20. Click on `Run workflow`.
21. Enter `release/6.8` for the release branch.
22. Enter the PR number you saved from step 16.
23. Run the workflow and ensure you see the same results as step 17.
24. Optionally you can test by marking the newly created PRs with the `6.8` milestone and squash and merge them. This time it should NOT trigger the cherry pick tool because these are created by the bot and there is a check for that. This is to prevent accidents.
25. Optionally you can test what it looks like when the cherry pick tool fails by running the manual workflow again and this time put a release branch that doesn't exists. And using the same PR number you saved from step 16. You should see a failing message in your Slack channel from the code freeze notification bot.

### Other information:




### FOR PR REVIEWER ONLY:
